### PR TITLE
Clone moveit2 in source docker file

### PR DESCRIPTION
**Pull request by tylerjw**
### Description

This should fix https://github.com/ros-planning/moveit2/issues/403

### CODE REVIEW

**Comment by tylerjw** [comment]
I previously tried using the docker `COPY` command but I realized that won't work nicely because the Dockerfile is not in the root of the repo.

**Review by JafarAbdi — COMMENTED** [review]
We should update https://github.com/ros-planning/moveit2/blob/main/.docker/ci/Dockerfile#L29 to fetch the dependencies of moveit2 also

**Comment by tylerjw** [comment]
> We should update https://github.com/ros-planning/moveit2/blob/main/.docker/ci/Dockerfile#L29 to fetch the dependencies of moveit2 also

I don't understand what you mean?  This uses git to clone the repo and the vcs to import the repos that are in the moveit2.repos file.  Are you talking about the rosdep dependencies?

**Comment by tylerjw** [comment]
Oh, you are talking about the CI image.  I should look at updating that one too.

File: .docker/ci-testing/Dockerfile [hunk 1]
```diff
@@ -10,9 +10,32 @@ MAINTAINER Robert Haschke rhaschke@techfak.uni-bielefeld.de
...
+    git clone https://github.com/ros-planning/moveit2.git src/moveit2 -b main && \
+    vcs import src < moveit2/moveit2.repos && \
+    #
+    # Download all dependencies of MoveIt
+    rosdep update && \
+    DEBIAN_FRONTEND=noninteractive \
+    rosdep install -y --from-paths src --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
+    # Remove the source code from this container
+    rm -rf src && \
+    #
```
[the remark this answers is not in the document]
Comment (tylerjw) [reply .docker/ci-testing/Dockerfile]: Yes, you are right, I didn't realize how this file worked.

**Comment by tylerjw** [comment]
I removed the changes to the ci-testing image as they weren't needed (as that one is based off the ci one).

**Pull request merged**